program/src/instructions/init_update_mint_proof_v2.rs

```diff
@@ -19,6 +19,7 @@ pub struct InitUpdateMintProofV2<'info> {
     #[account(mut)]
     pub payer: Signer<'info>,
 
+    // TODO: does it tho? where? I dont think we ever check if this is a Mint account
```

Comment:
Good catch; I actually changed this recently to not validate `Mint` accounts so that it supports non-token based standards like `mpl-core`. The way to think about whitelist is that it's a generic on-chain collection manager. You can manage lists of any kind of assets or items. Given more time and less backwards compatibility constraints I would refactor this to be `item`, `item_proof`, `init_update_item_proof`, etc.